tool/requote.js

```diff
@@ -457,44 +459,65 @@ module.exports = {
             help: 'BGP 47 locale tag of the language for parameter values',
             defaultValue: 'en-US'
         });
+        parser.addArgument('--output-errors', {
+            type: fs.createWriteStream,
+            help: 'If provided, examples which fail to be requoted are written in this file as well as being printed to stdout '
+        });
     },
 
     async execute(args) {
+
+        const promises = [];
+
+        let outputErrors = null;
+        const output = new DatasetStringifier();
+        promises.push(StreamUtils.waitFinish(output.pipe(args.output)));
+        if (args.output_errors) {
+            outputErrors = new DatasetStringifier();
+            promises.push(StreamUtils.waitFinish(outputErrors.pipe(args.output_errors)));
+        }
+
         readAllLines(args.input_file)
             .pipe(new DatasetParser({ contextual: args.contextual, preserveId: true }))
             .pipe(new Stream.Transform({
                 objectMode: true,
-
                 transform(ex, encoding, callback) {
                     try {
                         const [newSentence, newProgram] =
                             requoteSentence(ex.id, ex.context, ex.preprocessed, ex.target_code, args.mode,
                                 args.requote_numbers, args.handle_heuristics, args.param_locale);
                         ex.preprocessed = newSentence;
                         ex.target_code = newProgram;
-                        callback(null, ex);
+                        ex.is_ok = true;
+                        this.push(ex);
+
                     } catch(e) {
                         console.error('**************');
                         console.error('Failed to requote');
                         console.error(ex.id);
                         console.error(ex.preprocessed);
                         console.error(ex.target_code);
                         console.error('**************');
-                        if (args.skip_errors)
-                            callback(null);
-                        else
+                        ex.is_ok = false;
+                        if (args.skip_errors) {
+                            this.push(ex);
+                        } else {
                             callback(e);
+                        }
                     }
+                    callback();
```

Comment:
You're calling callback twice if skip_errors is not set - this will cause a crash.